### PR TITLE
Image gallery

### PR DESCRIPTION
**Pull request by jleiandy**
- Restructured directories
- Added react-icon, propTypes, styleloader, and cssloader node packages
- Added image carousel, WIP
- Configured Jest config to ignore CSS files, this required creating a styleMock.js file that exports an empty object.

### CODE REVIEW

File: client/components/App.jsx [hunk 1]
```diff
@@ -1,5 +1,11 @@
 import React from 'react';
+import ProductOverview from './ProductOverview/ProductOverview';
```
Comment (cschucode) [inline client/components/App.jsx:3]: Separate external imports from component imports by one line please.

File: client/components/ProductOverview/ImageGallery.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+import { AiOutlineArrowLeft, AiOutlineArrowRight } from 'react-icons/ai';
+import PropTypes from 'prop-types';
+
+const ImageGallery = ({ currentStyle }) => {
+  const [currentImageIndex, changeImage] = useState(0);
```
Comment (cschucode) [inline client/components/ProductOverview/ImageGallery.jsx]: For naming here, you'll want it to be something like `[currentImageIndex, setCurrentImageIndex]`.

File: client/components/ProductOverview/ProductOverview.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import mockProductStyles from '../../../MockData/ProductStyles';
+import ImageGallery from './ImageGallery';
+import '../../css/ProductOverviewCSS.css';
```
Comment (cschucode) [inline client/components/ProductOverview/ProductOverview.jsx:6]: I would separate the css import by one line just as a matter of style.

File: client/components/ProductOverview/ProductOverview.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+const ProductOverview = () => {
+  const currentStyleIndex = 0;
+  const currentStyle = mockProductStyles.results[currentStyleIndex];
+  return (
+    <>
```
Comment (cschucode) [inline client/components/ProductOverview/ProductOverview.jsx]: I'm not sure about this choice. I'm wondering if React.fragment would be better than empty markup?

File: client/tests/ProductOverview.test.jsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import { render, screen } from '@testing-library/react';
+// import userEvent from '@testing-library/user-event';
+
+import ProductOverview from '../components/ProductOverview/ProductOverview';
+
+beforeEach(() => {
+  render(<ProductOverview />);
+});
+
+describe('Product Overview Component', () => {
```
Comment (cschucode) [inline client/tests/ProductOverview.test.jsx:12]: Nice Tests!

File: jest.config.js [hunk 6]
```diff
@@ -86,7 +86,7 @@ module.exports = {
 
   // A map from regular expressions to module names or to arrays of module names that
   // allow to stub out resources with a single module
-  // moduleNameMapper: {},
+  moduleNameMapper: { '\\.(css|less)$': '<rootDir>/styleMock.js' },
```
Comment (cschucode) [inline jest.config.js:89]: does this get removed later or is this the workaround? Seems a little hacky. 🤔

File: client/components/App.jsx (hunk 1 above)
Comment (jleiandy) [reply client/components/App.jsx:3]: corrected.

File: client/components/ProductOverview/ImageGallery.jsx (hunk 2 above)
Comment (jleiandy) [reply client/components/ProductOverview/ImageGallery.jsx]: refactored.

File: client/components/ProductOverview/ProductOverview.jsx (hunk 3 above)
Comment (jleiandy) [reply client/components/ProductOverview/ProductOverview.jsx:6]: refactored.

File: client/components/ProductOverview/ProductOverview.jsx (hunk 4 above)
Comment (jleiandy) [reply client/components/ProductOverview/ProductOverview.jsx]: Swapped react.fragments to divs with more descriptive classname

File: jest.config.js (hunk 6 above)
Comment (jleiandy) [reply jest.config.js:89]: this is official documentation, https://jestjs.io/docs/webpack

**Review by cschucode — APPROVED** [review]
I'm cool with this. I'm sure there might be a few things that still will get refactored and polished up, but it seems okay for now.

**Pull request merged**